src/compiler/config/validate-service-worker.ts

```diff
@@ -54,16 +77,19 @@ export const validateServiceWorker = (config: d.ValidatedConfig, outputTarget: d
     outputTarget.serviceWorker.swSrc = join(config.rootDir, outputTarget.serviceWorker.swSrc);
   }
 
-  if (!isString(outputTarget.serviceWorker.swDest)) {
-    outputTarget.serviceWorker.swDest = join(outputTarget.appDir, DEFAULT_FILENAME);
```

Comment:
Is there a reason we're able to remove the `DEFAULT_FILENAME` assignment? I'm not sure I understand how/why we're able to combine these two `if` statements into one

Reply:
The [change on lines 53-55](https://github.com/ionic-team/stencil/pull/3813/files#diff-c2b7ab30b2bdcc180082ae16ba7b8aa1da392a53ea5054cea9bfb506b98d3fe0R53-R55) covers part of what we were doing here:

```ts
    swDest: isString(outputTarget.serviceWorker?.swDest)
      ? outputTarget.serviceWorker.swDest
      : join(outputTarget.appDir ?? '', DEFAULT_FILENAME),
```

I wanted to move it up in the file so that earlier in the function we were sort of declaring what a good default value for this looks like. As I see it, there's two things we want to do w/r/t the `swDest`:

1. ensure that if no `swDest` is set we provide a known-good default value (`join(outputTarget.appDir ?? '', DEFAULT_FILENAME`)
2. ensure that if a value is set for `swDest` that it is an absolute path, and if not adjust it to be so

it was a while since I opened the PR so tbh I don't remember exactly my chain of reasoning but I think I wanted to separate no. 1 from no. 2, putting no. 1 higher in the function body in one spot where we set default values if there is no input provided (we do that for `globDirectory` too).


Also the diff is presented slightly wrong here (on github) - what I actually did was delete this code:

https://github.com/ionic-team/stencil/blob/65444226303bf2ccb678576b4fb57b4df35e0a18/src/compiler/config/validate-service-worker.ts#L57-L59

and move it up top and then add an `isString` check to this bit:

https://github.com/ionic-team/stencil/blob/65444226303bf2ccb678576b4fb57b4df35e0a18/src/compiler/config/validate-service-worker.ts#L61-L63

which was just to fix a `strictNullChecks` error since the type of `.swDest` is `string | undefined`, so I wanted to add the `isString` typeguard because `isAbsolute` takes `string`.